src/CMakeLists.txt

```diff
@@ -17,7 +17,7 @@ endif()
 
 set(TRANSLATIONS_DIR "${PROJECT_SOURCE_DIR}/translations")
 
-find_package(Qt5 COMPONENTS Widgets Core Gui Xml Network Concurrent Multimedia Svg REQUIRED)
+find_package(Qt5 COMPONENTS Widgets Core Gui Xml Network Concurrent Multimedia Svg Sql REQUIRED)
```

Comment:
Do we need the sql dependency without the new hisotry patch?

Reply:
Yes, the OMEMO plugin uses QtSql, and if it's not linked to the main psi binary, on shutdown the QtSql library will be unloaded together with the OMEMO plugin, but before `qt_call_post_routines` gets a chance to perform a hook registered by QtSql. This will lead to a crash.

Reply:
So basically you are saying we can't link QtSql to just plugin. We have to link to both the plugin and Psi to avoid crashes even if Psi doesn't use it explicitly.

Reply:
Correct.

Reply:
As I understood the main binary file should be linked with the sql library to avoid crashes on plugin unloading. Am I right?

Reply:
Not exactly. The plugin unloads without a crash, but then later qt_call_post_routines tries to call a function that does not exist anymore (because QtSql was unloaded together with a plugin), and *that* leads to the crash.

Reply:
By the way, plugins get unloaded before shutdown, so the crash reproduces on every shutdown, not just when user disabled the plugin.

Reply:
Я в код плагина не вникал, но мне интересно, почему нельзя работу по финализации провести в методе disable(). Он вызывается каждый раз при выгрузке плагина.

Reply:
Плагин тут совсем ни при чём. Проблема в том, что архитектура Qt не подразумевает, что QtSql, будучи однажды загруженным, потом будет выгружаться.

 Можно провести эксперимент: создать новый плагин который в enable просто делает `QSqlDatabase::drivers()`. Всё, если QtSql не прилинкован к основному бинарнику Psi, то на выходе из Psi будет креш, так как QtSql внутри делает `qAddPostRoutine(cleanDriverDict)`. Если QtSql выгружается, то `cleanDriverDict` начинает указывать куда попало.

Reply:
Замечательно. В таком случае можно смержить. тем более, что так или иначе, но патч с новой историей скоро будет включен в основную ветку. Но есть одно НО. У нас основная система сборки qmake/qconf. Так что QtSql нужно будет добавить туда. Хотя, думаю, это уже и так есть в патче на новую историю. Также, когда дело дойдет до мержа плагина, я буду настаивать на поддержке qmake сборки (сейчас ее нет).

Reply:
> У нас основная система сборки qmake/qconf. Так что QtSql нужно будет добавить туда.

Так я вроде бы добавил: https://github.com/stigger/psi/blob/047910ee95647df7d6529401668e9924f9b930d7/src/src.pri#L1

> Также, когда дело дойдет до мержа плагина, я буду настаивать на поддержке qmake сборки (сейчас ее нет).

Сделаю.

Reply:
@stigger 
> Так я вроде бы добавил: 

Я имел ввиду qcm модуль для проверки наличия QtSql. Например, вот модуль для мультимедии:
 
https://github.com/psi-im/psi/blob/master/qcm/qtmultimedia.qcm